vm/src/stdlib/collections.rs

```diff
@@ -431,32 +384,32 @@ mod _collections {
         #[pymethod(magic)]
         fn contains(&self, needle: PyObjectRef, vm: &VirtualMachine) -> PyResult<bool> {
             let start_state = self.state.load();
-            let deque = self.borrow_deque().clone();
-            for element in deque.iter() {
-                let is_element = vm.identical_or_equal(element, &needle)?;
-
-                if start_state != self.state.load() {
-                    return Err(vm.new_runtime_error("deque mutated during iteration".to_owned()));
-                }
-                if is_element {
-                    return Ok(true);
-                }
+            let ret = self.mut_contains(vm, &needle)?;
+            if start_state != self.state.load() {
+                Err(vm.new_runtime_error("deque mutated during iteration".to_owned()))
+            } else {
+                Ok(ret)
             }
-
-            Ok(false)
         }
 
-        #[pymethod(magic)]
-        #[pymethod(name = "__rmul__")]
-        fn mul(&self, value: isize, vm: &VirtualMachine) -> PyResult<Self> {
-            let deque: SimpleSeqDeque = self.borrow_deque().into();
-            let mul = sequence::seq_mul(vm, &deque, value)?;
+        fn _mul(&self, n: isize, vm: &VirtualMachine) -> PyResult<VecDeque<PyObjectRef>> {
+            let deque = self.borrow_deque();
+            let n = vm.check_repeat_or_memory_error(deque.len(), n)?;
```

Comment:
bit confuse, should it return overflow error rather than memory error?

Reply:
after reading the issue 45044 from cpython, I think we should replace all the memory error with overflow error because currently we did not catch the malloc failing.
What do you think? @DimitrisJim @youknowone

Reply:
that sounds like reasonable